SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp

```diff
@@ -499,14 +506,7 @@ void UGlobalStateManager::AuthorityChanged(const Worker_AuthorityChangeOp& AuthO
 		}
 		case SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID:
 		{
-			// We can reach this point with bCanBeginPlay==true if the server
-			// that was authoritative over the GSM restarts.
-			if (!bCanBeginPlay)
-			{
-				BecomeAuthoritativeOverAllActors();
-				SetCanBeginPlay(true);
-			}
-
+			bCanSpawnWithAuthority = !bCanBeginPlay;
```

Comment:
This seems a little magic - could you add an explanation about why this is necessary?